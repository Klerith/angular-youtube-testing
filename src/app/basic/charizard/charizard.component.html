<h1>Pok√©mon</h1>
<div *ngIf="!charizard; else elseBlock">
    <h2>Loading...</h2>
</div>

<ng-template #elseBlock>
    <h3>{{ charizard?.name | titlecase }}</h3>
    <img [src]="charizard?.sprites?.front_default" [alt]="charizard?.name">
</ng-template>
