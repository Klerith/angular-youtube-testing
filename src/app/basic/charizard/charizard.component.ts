import { Component, OnInit } from '@angular/core';
import { Pokemon } from '../interfaces';
import { PokemonService } from '../services/pokemon.service';

@Component({
  selector: 'app-charizard',
  templateUrl: './charizard.component.html',
  styleUrls: ['./charizard.component.css']
})
export class CharizardComponent implements OnInit {

  public charizard?: Pokemon;

  constructor( private pokemonService: PokemonService ) { }

  ngOnInit(): void {

    this.pokemonService.getPokemon(6)
      .subscribe( pokemon => {
        this.charizard = pokemon;
        // console.log(pokemon);
      });

  }

}
