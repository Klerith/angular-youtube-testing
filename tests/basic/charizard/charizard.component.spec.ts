import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';

import { CharizardComponent } from '../../../src/app/basic/charizard/charizard.component';
import { PokemonService } from '../../../src/app/basic/services/pokemon.service';

describe('CharizardComponent', () => {
  let component: CharizardComponent;
  let fixture: ComponentFixture<CharizardComponent>;
  let compiled: HTMLElement;
  let service: PokemonService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ CharizardComponent ],
      imports: [ HttpClientTestingModule ],
      providers: [ PokemonService ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CharizardComponent);
    component = fixture.componentInstance;
    service = TestBed.inject( PokemonService );
    httpMock = TestBed.inject( HttpTestingController );
    
    fixture.detectChanges();
    compiled = fixture.nativeElement;

  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hacer match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();    
  });

  test('debe de mostrar un loading al inicio', () => {
    const h2 = compiled.querySelector('h2');
    expect( h2?.textContent ).toContain('Loading...');
  });

  test('debe de cargar a charizard inmediatamente', () => {

    const dummyPokemon = {
      name: 'charizardo!!',
      sprites: {
        front_default: 'https://charizard.com/sprite.png'
      }
    };

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6');
    expect( request.request.method ).toBe('GET');
    request.flush( dummyPokemon );

    fixture.detectChanges();
    // console.log(compiled.innerHTML);
    const h3  = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect( h3?.textContent?.toLocaleLowerCase() ).toContain( dummyPokemon.name.toLocaleLowerCase() );
    expect( img?.src ).toBe( dummyPokemon.sprites.front_default );
    expect( img?.alt ).toBe( dummyPokemon.name );


  });


});
