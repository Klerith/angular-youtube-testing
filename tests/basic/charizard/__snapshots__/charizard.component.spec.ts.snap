// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharizardComponent debe de hacer match con el snapshot 1`] = `
"<h1>Pok√©mon</h1><div><h2>Loading...</h2></div><!--bindings={
  \\"ng-reflect-ng-if-else\\": \\"[object Object]\\"
}--><!--container-->"
`;
